### PR TITLE
internal/mongodoc: add Contents field to mongodoc.Entity

### PR DESCRIPTION
**Pull request by rogpeppe**
This will allow us to gain quick access to selected files
in an entity's archive such as README and icon.svg.

We'll actually use the field in an upcoming PR.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/458/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/459/

File: internal/mongodoc/doc.go [hunk 1]
```diff
@@ -72,4 +72,30 @@ type Entity struct {
...
 	// and search ranking field(s).
+
+	// Contents holds entries for frequently accessed
+	// entries in the file's blob. Storing this avoids
+	// the need to linearly read the zip file's manifest
+	// every time we access one of these files.
+	Contents map[FileId]ZipFile `json:",omitempty" bson:",omitempty"`
+}
+
+type FileId string
```
Comment (jrwren) [inline internal/mongodoc/doc.go:83]: Why alias string here?

**Comment by jrwren** [comment]
LGTM with just a question. I look forward to seeing how Contents is used.
:+1:

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**